python/pyspark/rdd.py

```diff
@@ -188,6 +188,12 @@ def __init__(self, jrdd, ctx, jrdd_deserializer=AutoBatchedSerializer(PickleSeri
         self._id = jrdd.id()
         self.partitioner = None
 
+    def __enter__(self):
```

Comment:
Is that true? Doesn't it call `__enter__` on the instance of `rdd.cache().map(...)` where `is_cached` is set to False?

Quick verification:

```python
def __enter__(self)
    if self.is_cached:
        return self
    else:
        raise ValueError("r")

 with rdd.cache().map(lambda x: x) as t:
    pass
```

raises a `ValueError`